crates/runtime/src/table.rs

```diff
@@ -80,8 +179,19 @@ impl Table {
             return Err(Trap::wasm(ir::TrapCode::TableOutOfBounds));
         }
 
-        for i in start..end {
-            self.set(i, val.clone()).unwrap();
+        match val {
+            TableElement::FuncRef(r) => unsafe {
+                self.with_funcrefs_mut(move |elements| {
+                    let elements = elements.unwrap();
```

Comment:
I'll fix that.

Reply:
Fixed this in the `wasmtime` crate as returning a "table out of bounds" trap (the only trap code that would make sense to return from this runtime function) here for mismatched type seems wrong; this is an API bug anyway since mismatched type shouldn't pass module validation.